MaxText/checkpointing.py

```diff
@@ -92,34 +126,53 @@ def map_to_pspec(data, pspec):
     else:
       return type_handlers.RestoreArgs()
 
-  restore_args = jax.tree_util.tree_map(map_to_pspec,
-                                        abstract_unboxed_pre_state,
-                                        state_mesh_annotations)
-  latest_step = checkpoint_manager.latest_step()
-  if latest_step is not None:
-    max_logging.log(f"restoring state from this run's directory latest step \
-        {latest_step}")
-    return checkpoint_manager.restore(latest_step, abstract_unboxed_pre_state,
+  # Set restore_args based whether to load data iterator
```

Comment:
I'm very stressed by this code and don't feel comfortable pushing this in our "simple" reference for customers.